spitfire/runtime/udn.py

```diff
@@ -134,23 +135,31 @@ def _resolve_udn_prefer_attr3(_object, name, raise_exception=False):
 def _resolve_placeholder(name, template, global_vars):
     placeholder_cache = template.placeholder_cache
     if placeholder_cache and name in placeholder_cache:
-        return placeholder_cache[name]
+        ph = placeholder_cache[name]
+        if isinstance(ph, weakref.ReferenceType) and ph() is not None:
```

Comment:
I think it may be cleaner to write it like:
```
if isinstance(ph, weakref.ReferenceType):
  v = ph()
  if ph is not None:
    return v
else:
  return ph
```

This way you avoid invoking ph() twice. You also get to use an else as opposed to an elif.

Also, why are we not returning when ph() is None?